qualtran/resource_counting/classify_bloqs.py

```diff
@@ -109,7 +111,62 @@ def classify_t_count_by_bloq_type(
     return classified_bloqs
 
 
-def bloq_is_clifford(b: Bloq):
+_CLIFFORD_ANGLES = np.array(
+    [
+        np.pi,
+        np.pi / 2,
+        3 * np.pi / 2,
+        -np.pi,
+        -np.pi / 2,
+        -3 * np.pi / 2,
+        0.0,
+        -2 * np.pi,
+        2 * np.pi,
+    ]
+)
+_CLIFFORD_EXPONENTS = np.array([1.0, 0.5, 1.5, -1.0, -0.5, -1.5, 0.0, -2, 2])
+_T_ANGLES = np.array([np.pi / 4, -np.pi / 4])
+_T_EXPONENTS = np.array([0.25, -0.25])
+
+
+def bloq_is_t_gate(b: Bloq) -> bool:
+    """Whether a bloq should be counted as a T gate.
+
+    This will return `True` for any instance of `TGate`. It will also consider
+    single-qubit rotations and return True if the angle corresponds to a T gate
+    (up to clifford reference frame).
+    """
+    from qualtran.bloqs.basic_gates import Rx, Ry, Rz, TGate, XPowGate, YPowGate, ZPowGate
+
+    if isinstance(b, TGate):
+        return True
+
+    if isinstance(b, (Rz, Rx, Ry)):
+        if is_symbolic(b.angle):
+            return False  # Symbolic rotation
+        if np.any(np.abs(b.angle - _T_ANGLES) < b.eps):
```

Comment:
`bloq.eps` is currently not propagated properly, and in most places uses hardcoded values (usually 1e-11). Might be better to check against a configurable precision?

Reply:
comparing against a classical float-comparison value. If users want to "round" angles that should best be done by an external force. This tolerance is just for classical, python float comparison. Thanks!